### PR TITLE
Ms 311 implement videolayout

### PR DESCRIPTION
**Pull request by jerzywilczek**
Reviewers are encouraged to read all commit messages before reviewing the changes, as the commit messages explain what specifically was done in this PR.

### CODE REVIEW

**Review by Karolk99 — APPROVED** [review]
VideoLayout look good and have all necessary features from LIVE point of view.

**Review by WojciechBarczynski — CHANGES_REQUESTED** [review]

File: native/membrane_videocompositor/src/compositor.rs [hunk 1]
```diff
@@ -350,26 +350,57 @@ impl State {
...
-                );
-            }
+        );
+
+        Ok(())
+    }
+
+    pub fn update_properties(
+        &mut self,
+        idx: usize,
```
[the remark this answers is not in the document]
Comment (jerzywilczek) [reply native/membrane_videocompositor/src/compositor.rs:377]: I don't think so. This is the type canonically used for indexing, being an index is its only purpose. Why would we use u32? It doesn't bring any benefits.

File: native/membrane_videocompositor/src/lib.rs [hunk 2]
```diff
@@ -159,32 +160,91 @@ fn force_render(
...
+    )?;
+
+    Ok(atoms::ok())
+}
+
+fn convert_z(z: f32) -> f32 {
+    // we need to do this because 0.0 is an intuitively standard value and maps onto 1.0,
+    // which is outside of the wgpu clip space
+    1.0 - z.max(1e-7)
+}
```
[the remark this answers is not in the document]
Comment (jerzywilczek) [reply native/membrane_videocompositor/src/lib.rs:203]: I've decided to leave this as is

**Comment by Qizot** [comment]
@WojciechBarczynski Don't think that much 😉

**Review by WojciechBarczynski — APPROVED** [review]
Great job :)

File: native/membrane_videocompositor/src/compositor.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (WojciechBarczynski) [reply native/membrane_videocompositor/src/compositor.rs:377]: ok

**Pull request merged**